### ISSUE TITLE
Create minishift-data.json to persists data specific to minishift instance

### ISSUE THREAD

**Issue by budhrg** [minishift/minishift#487]
We need `minishift-data.json` to persists few data specific to minishift instance like 'oc' version etc.

Currently, we need following data as:
```
{
  "ocVersion": "1.4.1",
  "ocPath": "/home/budhram/.minishift/cache/oc/v1.4.1/oc",
  "isRegistered": false 
}
``` 

`isRegistered` is updated when we register a VM, so that we can read that in stop/delete instead of having to ssh into the VM (which might be in an error state) and `ocVersion` and `ocPath` are needed to make sure we are using correct oc version on start/stop/delete for instance. 

This is a requirement for #342 and other issues.

@hferentschik Please add if I miss anything here.

**Comment by hferentschik** [comment]
In code we need utility methods to read/write the config file and parse it into a types config struct of sorts.

**Comment by gbraad** [comment]
@hferentschik @budhrg This is something I also need for the hostfolders.

**Comment by budhrg** [comment]
As a first cut of it, we agree to have minimal config.json file with :
```
{
  "ocPath": "/home/budhram/.minishift/cache/oc/v1.4.1/oc"
}
```
As we go further, we can add more properties.

**Comment by gbraad** [comment]
I think the use of 'MinishiftData' confuses as it has not much to do with the actual data for minishift (such as the volume files for an instance). It is better to call this something with 'instance' in the naming.

**Comment by hferentschik** [comment]
> I think the use of 'MinishiftData' confuses as it has not much to do with the actual data for minishift (such as the volume files for an instance). It is better to call this something with 'instance' in the naming.

+1 especially once we can manage multiple instances

**Comment by hferentschik** [comment]
Merged via pull request #521
